src/common/cluster-store.ts

```diff
@@ -182,10 +196,20 @@ export class ClusterStore extends BaseStore<ClusterStoreModel> {
     return this.clusters.get(id);
   }
 
+  getDeadById(id: ClusterId): [ClusterModel, LoadKubeError] {
+    return this.deadClusters.get(id);
+  }
+
   getByWorkspaceId(workspaceId: string): Cluster[] {
-    const clusters = Array.from(this.clusters.values())
-      .filter(cluster => cluster.workspace === workspaceId);
-    return _.sortBy(clusters, cluster => cluster.preferences.iconOrder);
+    const clusters = this.clustersList.filter(c => c.workspace === workspaceId);
+    return _.sortBy(clusters, c => c.preferences?.iconOrder);
+  }
+
+  getCountInWorkspace(workspaceId: string): number {
```

Comment:
It's not obvious that this counts also dead and disabled clusters? Maybe should not be part of ClusterStore API?

If we really want to keep it here then name should be `getVisibleCountInWorkspace` and it should not count disabled clusters.